docs/query-help-style-guide.md

```diff
@@ -0,0 +1,202 @@
+
+### Location and file name
+
+Query help files must have the same base name as the query they describe and must be located in the same directory.  
+
+### File structure and layout
+
+Query files are written using an XML format called Qhelp, and stored in a file with a `.qhelp` extension. The basic structure is as follows:
+
+```
+<!DOCTYPE qhelp SYSTEM "qhelp.dtd">
+<qhelp>
+    CONTAINS one or more section-level elements
+</qhelp>
+```
+
+The header and single top-level `qhelp` element are both mandatory. 
+
+### Section-level elements
+
+Section-level elements are used to group the information within the query help file. All query help files should include at least the following section elements, in the order specified:
+
+1.  `overview`—a summary of the purpose of the query, including an explanation of what the code does and the how the behavior of the program is affected.
+2.  `recommendation`—information on how to fix the issue highlighted by the query.
+3.  `example`—an example of code showing the problem. Where possible, this section should also include a solution to the issue.
+4.  `references`—relevant references to best practice. 
+
+For further information about the other section-level, block, list and table elements supported by the qhelp format, see [Qhelp files](https://wiki.semmle.com/display/SD/Qhelp+files).
+
+
+## English style
+
+You should write the overview and recommendation elements in simple English that is easy to follow. You should:
+
+*   Use simple sentence structures and avoid complex or academic language.
+*   Avoid colloquialisms and contractions.
+*   Use US English spelling throughout.
+*   Use words that are in common usage.
+
+## Code examples
+
+Whenever possible, you should include a code example that helps to explain the issue you are highlighting. Any code examples that you include should adhere to the following guidelines:
+
+*   The example should be less than 20 lines and ideally it should be runnable. If this is not possible, ensure that the example very clearly illustrates the issue that you are highlighting.
+*   Put the code example after the recommendation element where possible. Only include an example in the description element if absolutely necessary.
+*   If you are using an example to illustrate the solution to a problem, and the change required is minor, avoid repeating the whole example. It is preferable to either describe the change required or to include a smaller snippet of the corrected code.
+*   Clearly indicate which of the samples is an example of bad coding practice and which is recommended practice.
+
+There are several ways of including a code example in a query help file:
+
+*   Directly include the code as nested text in the `example` section, Use the `sample` block and define a `language` attribute to specify the language of the example:
+
+```       
+<example>
+<p>This example highlights poor coding practice</p>
+<sample language = "java" />
+
+Example of poor code
```

Comment:
This doesn't seem properly nested.  There's no code in the `<sample />`.  Also, aren't we usually recommending to always have the example code snippets in their own files and include them with the `src="file.java"` property as below?

Reply:
I suggested the nested code approach after reading the 'Block elements' table in this topic: https://help.semmle.com/wiki/display/SD/Qhelp+files – and since I could find any examples of qhelp in out built-in queries the nesting is likely to be incorrect! 

Happy to delete this part if specifying `src` is the preferred style.